sklearn/utils/_indexing.py

```diff
@@ -324,7 +324,11 @@ def _safe_indexing(X, indices, *, axis=0):
     if (
         axis == 1
         and indices_dtype == "str"
-        and not (_is_pandas_df(X) or _use_interchange_protocol(X))
+        and not (
+            _is_pandas_df(X)
+            or _is_polars_df_or_series(X)
+            or _use_interchange_protocol(X)
+        )
```

Comment:
Please update an existing test for `safe_indexing` or add a new one to test this outside of the `permutation_importance` use case.